### ISSUE TITLE
Search Bar Icon lacks automatic color blending

### ISSUE THREAD

**Issue by NathanJPlummer** [aashishnagpal/ba-ui-framework-eureka#3]
Right now we have a predefined fill color for our search icon.  This works great at the moment, but if we change the color scheme at a later point we'll have to replace the icon- or at least change its hex values to match.

I know GIMP has it color blend algorithms available publicly- somewhere. 

Combing this information with SASS's ability to perform calculations on Hex Values, we could set the color blend to be automatic.

That's something I think I'll wait on until after Hackathon though.

For reference, in the mockup the color blend was achieved by using the original icon colors and a "screen" blend mode.

**Comment by aashishnagpal** [comment]
Hi Nathan @NathanJPlummer 

Could you please help me understand this better? I guess CSS3 has something of `color-blend-mode` sorts. Understanding what you want to achieve might help me point you in the right direction.

**Comment by NathanJPlummer** [comment]
@aashishnagpal 

Sorrry for the delay, just getting back from work.

I think you maybe already did point me in the right direction.  Did not realize this function was already built into CSS.  My intent was to find the algorithm that GIMP uses and throw it into SCSS.  Managed to find it just fine, but while SCSS will let you do calculations on integers and it will let you do calculations on hex values, it won't let you mix both.

Blend modes are ways of creating different effects when one image- or element- is on top of another.  In the case of the search bar, during the design phase, I took the magnify glass icon at it's original color, layered it over the blue search bar, and chose "screen" blend mode.  This creates a  blue shadow for our icon with enough contrast to make it unique.

To replicate this in the actual code I just just took the color of the blend output and changed the magnifying glass's color to that.  This works as long as our color scheme stays the same, but without using a blend mode if we decided to change our color scheme, the icon will remain blue.

Anyway, I'll try this out now with the mix-blend-mode you pointed out.

Edit:

And.... done.
